beetsplug/mpdstats.py

```diff
@@ -118,12 +121,16 @@ def currentsong(self):
         """Return the path to the currently playing song, along with its
         songid.  Prefixes paths with the music_directory, to get the absolute
         path.
+        In some cases, we need to remove the local path from MPD server,
+        we replace 'strip_path' with ''.
+        `strip_path` defaults to ''.
         """
         result = None
         entry = self.get('currentsong')
         if 'file' in entry:
             if not is_url(entry['file']):
-                result = os.path.join(self.music_directory, entry['file'])
+                file = entry['file'].replace(self.strip_path, '')
```

Comment:
Is it perhaps worrisome that this will replace *all* occurrences of `strip_path` from the string—not just the leftmost one?

Reply:
Added count=1 to replace to stop at first occurence

Reply:
I don't think this will guarantee the first occurrence still? I think something like the following is probably best:

```Python
file = entry['file']

if file.startswith(self.strip_path):
    file = file[len(self.strip_path):]
```

It seems like Python 3.9+ has `removeprefix` but I think we still support <3.9.

Reply:
Yes indeed, thanks @jackwilsdon! A little utility function that behaves like `removeprefix` may be in order.

Reply:
I see your point, indeed a safer approach

Reply:
I changed the code accordingly, please check latest commit